pkg/kubelet/nodelease/controller.go

```diff
@@ -155,18 +157,44 @@ func (c *controller) retryUpdateLease(base *coordv1beta1.Lease) {
 // newLease constructs a new lease if base is nil, or returns a copy of base
 // with desired state asserted on the copy.
 func (c *controller) newLease(base *coordv1beta1.Lease) *coordv1beta1.Lease {
+	// Use the bare minimum set of fields; other fields exist for debugging/legacy,
+	// but we don't need to make node heartbeats more complicated by using them.
 	var lease *coordv1beta1.Lease
 	if base == nil {
-		lease = &coordv1beta1.Lease{}
+		lease = &coordv1beta1.Lease{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      c.holderIdentity,
+				Namespace: corev1.NamespaceNodeLease,
+			},
+			Spec: coordv1beta1.LeaseSpec{
+				HolderIdentity:       pointer.StringPtr(c.holderIdentity),
+				LeaseDurationSeconds: pointer.Int32Ptr(c.leaseDurationSeconds),
+			},
+		}
 	} else {
 		lease = base.DeepCopy()
 	}
-	// Use the bare minimum set of fields; other fields exist for debugging/legacy,
-	// but we don't need to make node heartbeats more complicated by using them.
-	lease.Name = c.holderIdentity
-	lease.Spec.HolderIdentity = pointer.StringPtr(c.holderIdentity)
-	lease.Spec.LeaseDurationSeconds = pointer.Int32Ptr(c.leaseDurationSeconds)
 	lease.Spec.RenewTime = &metav1.MicroTime{Time: c.clock.Now()}
+
+	// Setting owner reference needs node's UID. Note that it is different from
+	// kubelet.nodeRef.UID. When lease is initially created, it is possible that
+	// the connection between master and node is not ready yet. So try to set
+	// owner reference every time when renewing the lease, until successful.
+	if lease.OwnerReferences == nil || len(lease.OwnerReferences) == 0 {
+		if node, err := c.client.CoreV1().Nodes().Get(c.holderIdentity, metav1.GetOptions{}); err == nil {
+			lease.OwnerReferences = []metav1.OwnerReference{
+				{
+					APIVersion: corev1.SchemeGroupVersion.WithKind("Node").Version,
+					Kind:       corev1.SchemeGroupVersion.WithKind("Node").Kind,
+					Name:       c.holderIdentity,
+					UID:        node.UID,
+				},
+			}
+		} else {
+			glog.Infof("failed to get node %q when trying to set owner ref to the node lease: %v", c.holderIdentity, err)
```

Comment:
Why is this `Infof` as opposed to `Errorf`?

Reply:
Done